src/plugins/core/cell.ts

```diff
@@ -457,20 +460,98 @@ export class CellPlugin extends CorePlugin<CoreState> implements CoreState {
     }
 
     const cellId = before?.id || this.getNextUid();
-    const didContentChange = hasContent;
-    const properties = { format, style };
-    const cell = this.createCell(cellId, afterContent, properties, sheetId);
-    if (before && !didContentChange && cell.isFormula()) {
-      // content is not re-evaluated if the content did not change => reassign the value manually
-      // TODO this plugin should not care about evaluation
-      // and evaluation should not depend on implementation details here.
-      // Task 2813749
-      cell.assignEvaluation(lazy(before.evaluated));
-    }
+    const cell = this.createCell(cellId, afterContent, format, style, sheetId);
     this.history.update("cells", sheetId, cell.id, cell);
     this.dispatch("UPDATE_CELL_POSITION", { cellId: cell.id, col, row, sheetId });
   }
 
+  private createCell(
+    id: UID,
+    content: string,
+    format: Format | undefined,
+    style: Style | undefined,
+    sheetId: UID
+  ): Cell {
+    style = isMarkdownLink(content) ? { textColor: LINK_COLOR, ...style } : style;
```

Comment:
why only markdown links?  If I were to write a formula which results in a hyperlink, then the builder will mark it as such , the linkDisplay will activate, but I won't have style of a link cell. Is this on purpose ?

Reply:
Euh non, not on purpose